lib/msf/core/post/linux/system.rb

```diff
@@ -102,6 +102,216 @@ def get_sysinfo
+    info = {}
+    orig = cmd_exec("cat /proc/cpuinfo").to_s
+    cpuinfo = orig.split("\n\n")[0]
+    # This is probably a more platform independent way to parse the results (compared to splitting and assigning preset indices to values)
+    cpuinfo.split("\n").each do |l|
+      info[:speed_mhz]   = l.split(': ')[1].to_i if l.include? 'cpu MHz'
+      info[:product]     = l.split(': ')[1]      if l.include? 'model name'
+      info[:vendor]      = l.split(': ')[1]      if l.include? 'vendor_id'
+    end
+    info[:cores] = orig.split("\n\n").size
+    info
+  rescue
+    raise "Could not get CPU information"
+  end
+
+  #
+  # Gets the hostname of the system
+  # @return [String]
+  #
+  def get_hostname
+    cmd_exec('uname -n').to_s
+  rescue
+    raise 'Unable to retrieve hostname'
+  end
+
+  #
+  # Gets the name of the current shell
+  # @return [String]
+  #
+  def get_shell_name
+    psout = cmd_exec('ps -p $$').to_s
+    psout.split("\n").last.split(' ')[3]
+  rescue
+    raise 'Unable to gather shell name'
+  end
+
+  #
+  # Checks if the system has gcc installed
+  # @return [Boolean]
+  #
+  def has_gcc?
+    command_exists? 'gcc'
+  rescue
+    raise 'Unable to check for gcc'
+  end
+
+  #
+  # Gets the version of gcc
+  # @return [String]
+  #
+  def gcc_version
+    return cmd_exec('gcc --version | head -n 1').to_s.split(' ')[3] if has_gcc?
+    raise 'System does not have gcc installed'
+  rescue
+    raise 'Unable to get gcc version'
+  end
+
+  #
+  # Checks if the `cmd` is installed on the system
+  # @return [Boolean]
+  #
+  def command_exists?(cmd)
+    cmd_exec("command -v #{cmd} && echo true").to_s.include? 'true'
+  rescue
+    raise "Unable to check if command `#{cmd}` exists"
+  end
+
+  #
+  # Gets the process id(s) of `program`
+  # @return [Array]
+  #
+  def pidof(program)
+    pids = []
+    full = cmd_exec('ps aux').to_s
+    full.split("\n").each do |pid|
+      pids << pid.split(' ')[1].to_i if pid.include? program
+    end
+    pids
+  end
+
+  #
+  # Checks if `file_path` is mounted on a noexec mount point
+  # @return [Boolean]
+  #
+  def noexec?(file_path)
+    mount = cmd_exec('cat /proc/mounts').to_s
+    mount_path = get_mount_path(file_path)
+    mount.lines.each do |l|
+      return true if l =~ Regexp.new("#{mount_path} (.*)noexec(.*)")
+    end
+    false
+  rescue
+    raise 'Unable to check for noexec volume'
+  end
+
+  #
+  # Checks if `file_path` is mounted on a nosuid mount point
+  # @return [Boolean]
+  #
+  def nosuid?(file_path)
+    mount = cmd_exec('cat /proc/mounts').to_s
+    mount_path = get_mount_path(file_path)
+    mount.lines.each do |l|
+      return true if l =~ Regexp.new("#{mount_path} (.*)nosuid(.*)")
+    end
+    false
+  rescue
+    raise 'Unable to check for nosuid volume'
+  end
+
+  #
+  # Checks for protected hardlinks on the system
+  # @return [Boolean]
+  #
+  def protected_hardlinks?
+    read_file('/proc/sys/fs/protected_hardlinks').to_s.strip.eql? '1'
```

Comment:
You can use `blank?` if you're one of _those_ people.

Reply:
Oh, nevermind. You're doing a comparison. Carry on.

Reply:
Subtle troll is subtle?

Reply:
No, but not to worry, you have a few years to refine your technique.

![missing blank](https://user-images.githubusercontent.com/434827/38883559-a15ceb02-42b0-11e8-84f1-0487a3c42e4d.png)